core/src/main/java/com/yahoo/oak/EntrySet.java

```diff
@@ -152,6 +130,25 @@
         this.keySerializer = keySerializer;
         this.valueSerializer = valueSerializer;
         this.valOffHeapOperator = valOffHeapOperator;
+
+        /*
+         * Key/value reference encoding (64 bit):
+         *     LSB |     offset     |     length     | block  | MSB
+         *         |     23 bit     |     23 bit     | 18 bit |
+         * Key/value upper limits (default):
+         *   -       block size: 256MB (28 bit)
+         *                       This is the default size, but might be larger/smaller.
+         *   -      data length: 256MB (28 bit)
+         *                       The maximal data length matches the size of the block.
+         *   - number of blocks: 256 (8 bit)
+         *                       The remaining number of bits
+         *   => number of allocated 1K items: ~128 million items (2^26)
+         *
+         * TODO: block size should be derived directly from the memory manager.
```

Comment:
I agree that the reference structure should be derived from memory manager. But is BlockPool is being used or not is up to memory manager. BlockPool can be its integrated part.

Reply:
We don't have the bigger picture because we haven't discussed yet the details about the memory management package.
In general, I believe that the `MemoryManager` itself should be responsible for encoding the reference.
But for now, I think that `MemoryManager` should at least have an additional API method that returns the maximal block size: `int getMaximalBlockSize()`. This way we can avoid this issue for now.

Reply:
Bottom line we should deal with it later. But better remove the line speaking about BlockPool

Reply:
I removed the TODO, but I think it is necessary to address that in the future. So at least note that in your Oak TODO list.